controllers/etcd_controller.go

```diff
@@ -251,6 +271,42 @@ func (r *EtcdReconciler) reconcile(ctx context.Context, etcd *druidv1alpha1.Etcd
 func (r *EtcdReconciler) delete(ctx context.Context, etcd *druidv1alpha1.Etcd) (ctrl.Result, error) {
 	logger.Infof("Deletion timestamp set for etcd: %s", etcd.GetName())
 
+	svc, err := r.getService(ctx, etcd)
+	if err != nil {
+		logger.Error(err, "error getting service")
+		return ctrl.Result{
+			Requeue: true,
+		}, err
+	}
+
+	ss, err := r.getStatefulSet(ctx, etcd)
+	if err != nil {
+		logger.Error(err, "error getting statefulset")
+		return ctrl.Result{
+			Requeue: true,
+		}, err
+	}
+
+	pvcList := &v1.PersistentVolumeClaimList{}
+	if ss != nil {
+		pvcList, err = r.listPersistentVolumeClaims(ss)
+		if err != nil {
+			return ctrl.Result{
+				Requeue: true,
+			}, err
+		}
+	}
+
+	if svc != nil && len(pvcList.Items) > 0 {
+		snapshotCtx, cancel := context.WithTimeout(ctx, DefaultTimeout)
+		defer cancel()
+		if err := r.takeFullSnapshot(snapshotCtx, etcd, svc); err != nil {
```

Comment:
Are we cutting off the ingress traffic to etcd before this?

Reply:
Not yet in this PR. From gardeners CP perspective, this will happen since service will be brought down first. From stand alone etcd-druid's perspective, i was suggesting to reuse healthz controller logic from multi-node etcd pr later.